backend/internal/app/handlers.go

```diff
@@ -8,23 +8,50 @@ package app
 
 import (
 	"encoding/json"
+	"fmt"
 	"io/ioutil"
 	"log"
 	"net/http"
-	"os"
 
 	"github.com/gin-gonic/gin"
 )
 
-func GetClientId(c *gin.Context) {
-	var id struct {
-		Client_id string `json:"client_id"`
+// Redirects the user to gamma authentication page
+func HandleLogin(c *gin.Context) {
+	c.Redirect(http.StatusPermanentRedirect,
+		fmt.Sprintf("%s?response_type=code&client_id=%s&redirect_uri=%s",
+			client.Endpoint.AuthURL,
+			client.ClientID,
+			client.RedirectURL))
+}
+
+// Removes the authentication cookie and makes the user unauthenticated
+func HandleLogout(c *gin.Context) {
+	c.SetCookie("suggestit", "", -1000, "/", c.Request.Host, true, true)
+}
+
+// Exchanges Gamma authentication code with access token
+// The cookie will be sent along with all further requests
+func HandleAuthenticationWithCode(c *gin.Context) {
+	code := c.Query("code")
+
+	token, err := getToken(code)
+	if err != nil {
+		log.Println(err)
+		c.AbortWithError(http.StatusUnauthorized, err)
+		return
 	}
-	id.Client_id = os.Getenv("CLIENT_ID")
 
-	c.JSON(http.StatusOK, id)
+	c.SetCookie("suggestit",
+		token.AccessToken,
+		24*60*60,
+		"/",
+		c.Request.Host,
+		true,
+		true)
```

Comment:
HttpOnly and secure is used when creating cookies but it does not use the correct domain, I will fix this

Reply:
Why are you then using getCookie in the frontend?

Reply:
I am not, those rows has been removed and now only the backend can access the cookie.